### PR TITLE
Upgrade jetty to 9.4.2

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
verifier run looks good.

### CODE REVIEW

File: http-server/src/main/java/io/airlift/http/server/HashLoginServiceProvider.java [hunk 1]
```diff
@@ -42,18 +37,9 @@ public HashLoginServiceProvider(HttpServerConfig config)
...
-            if (!isNullOrEmpty(authConfig)) {
-                HashLoginService service = new HashLoginService(HttpServerModule.REALM_NAME, authConfig);
-                service.loadUsers();
-                return service;
-            }
-            return null;
+        if (!isNullOrEmpty(authConfig)) {
+            return new HashLoginService(HttpServerModule.REALM_NAME, authConfig);
         }
-        catch (IOException e) {
```
Comment (electrum) [inline http-server/src/main/java/io/airlift/http/server/HashLoginServiceProvider.java]: Does this no longer throw `IOException`? Or throw at all?

File: http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java [hunk 2]
```diff
@@ -1370,10 +1370,10 @@ public ConnectionPoolDistribution(HttpClient httpClient, Processor processor)
                 Distribution distribution = new Distribution();
                 httpClient.getDestinations().stream()
                         .filter(PoolingHttpDestination.class::isInstance)
-                        .map(destination -> (PoolingHttpDestination<?>) destination)
+                        .map(destination -> (PoolingHttpDestination) destination)
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java]: Do this as
```java
.map(PoolingHttpDestination.class::cast)

File: http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java [hunk 3]
```diff
@@ -1370,10 +1370,10 @@ public ConnectionPoolDistribution(HttpClient httpClient, Processor processor)
...
                         .map(PoolingHttpDestination::getConnectionPool)
                         .filter(pool -> pool != null)
-                        .forEach(pool -> processor.process(distribution, pool));
+                        .forEach(pool -> processor.process(distribution, (DuplexConnectionPool) pool));
```
Comment (electrum) [inline http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java]: Add a separate map before this to do the cast

File: http-server/src/main/java/io/airlift/http/server/HashLoginServiceProvider.java (hunk 1 above)
Comment (nezihyigitbasi) [inline http-server/src/main/java/io/airlift/http/server/HashLoginServiceProvider.java]: `loadUsers()` was throwing previously and it's removed in this release, so nothing is throwing an IOE now. BTW it was already a noop in `9.3.13.M0`.

File: http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java (hunk 2 above)
Comment (nezihyigitbasi) [inline http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java]: I wanted to make minimal changes for the upgrade so I didn't change many of these `map` calls (there are many `map` calls that I can change like that). Now you asked for that I will do it shortly.

**Comment by nezihyigitbasi** [comment]
updated.

**Comment by nezihyigitbasi** [comment]
Superseded by #528.

**Pull request closed without merging**